setup.py

```diff
@@ -1,54 +1,72 @@
-#!/usr/bin/env python
-"""Setup file for python-matlab-bridge"""
+# -*- coding: utf-8 -*-
+"""
+Pymatbridge: A python interface to call out to Matlab(R)
+"""
 
 import os
 import sys
-import shutil
-import glob
+import filecmp
+import itertools
+import platform
 
+try:
+    import pkg_resources
+except ImportError:
+    import ez_setup
+    ez_setup.use_setuptools()
 
-# BEFORE importing distutils, remove MANIFEST. distutils doesn't properly
-# update it when the contents of directories change.
-if os.path.exists('MANIFEST'):
-    os.remove('MANIFEST')
+from setuptools import  setup, find_packages
+from setuptools.command.test import test as TestCommand
+from distutils import file_util
+from setuptools.command.build_ext import build_ext
 
+from pymatbridge.messenger import build_matlab, get_messenger_dir
 
-from distutils.core import setup
+from version import __version__, __build__, __release__
 
+messenger = pkg_resources.resource_filename('pymatbridge.messenger', get_messenger_dir())
+matlab    = pkg_resources.resource_filename('pymatbridge', 'matlab')
+newfiles = filecmp.cmpfiles(messenger, matlab, os.listdir(messenger), shallow=False)[1:]
 
-# Find the messenger binary file(s) and copy it to /matlab folder.
-from messenger.make import get_messenger_dir
-messenger_dir = get_messenger_dir()
+for binary in itertools.chain(*newfiles):
+    cmd = (os.path.join(messenger, binary), os.path.join(matlab, binary))
+    print('Copying %s' % binary)
+    file_util.copy_file(*cmd, update=True)
 
-for f in glob.glob("./messenger/%s/messenger.*" % messenger_dir):
-    shutil.copy(f, "./pymatbridge/matlab")
+class NoseTestCommand(TestCommand):
 
+    def finalize_options(self):
+        TestCommand.finalize_options(self)
+        self.test_args = []
+        self.test_suite = True
 
-# Get version and release info, which is all stored in pymatbridge/version.py
-ver_file = os.path.join('pymatbridge', 'version.py')
-exec(open(ver_file).read())
+    def run_tests(self):
+        import nose
+        args = 'nosetests -v --exe --with-cov '
+        if sys.version_info == (2, 7):
+            args += '--cover-package pymatbridge'
```

Comment:
Coverage report should not be mandatory on some versions. Instead it should be an option. That is

    python setup.py test --with-coverage 

should achieve this effect. Is that possible to do with this? 

Notice also that the coverage report on travis seems to be doing something wrong. We are getting a coverage report on some elements of IPython, instead of pymatbridge. Could you take a look at that?